pkg/asset/agent/configimage/configiso.go

```diff
@@ -0,0 +1,159 @@
+package configimage
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/vincent-petithory/dataurl"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent/image"
+)
+
+const (
+	configImageFilename = "agentconfig.iso"
+
+	pullSecretManifest          string = "/etc/assisted/manifests/pull-secret.yaml" //nolint:gosec // not hardcoded credentials
+	nmStateConfigManifest       string = "/etc/assisted/manifests/nmstateconfig.yaml"
+	clusterDeploymentManifest   string = "/etc/assisted/manifests/cluster-deployment.yaml"
+	clusterImageSetManifest     string = "/etc/assisted/manifests/cluster-image-set.yaml"
+	agentClusterInstallManifest string = "/etc/assisted/manifests/agent-cluster-install.yaml"
+	extraManifests              string = "/etc/assisted/extra-manifests"
+	hostConfigFiles             string = "/etc/assisted/hostconfig"
+	hostNameFiles               string = "/etc/assisted/hostnames"
+	kubeadminPasswordHash       string = "/opt/agent/tls/kubeadmin-password.hash" //nolint:gosec // not hardcoded credentials
+	kubeconfigTLSCerts          string = "/opt/agent/tls/admin-kubeconfig-signer"
+	kubeapiserverTLSCerts       string = "/opt/agent/tls/kube-apiserver"
+)
+
+// TODO - need to add rendezvous host file when available.
+var fileFilters = []string{
+	pullSecretManifest,
+	nmStateConfigManifest,
+	clusterDeploymentManifest,
+	clusterImageSetManifest,
+	agentClusterInstallManifest,
+	extraManifests,
+	hostConfigFiles,
+	hostNameFiles,
+	kubeadminPasswordHash,
+	kubeconfigTLSCerts,
+	kubeapiserverTLSCerts,
+}
+
+// ConfigImage is an asset that generates a configuration ISO that can be used configure hosts.
+type ConfigImage struct {
+	tmpPath string
+}
+
+var _ asset.WritableAsset = (*ConfigImage)(nil)
+
+// Dependencies returns the assets on which the Config asset depends.
+func (a *ConfigImage) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&image.Ignition{},
+	}
+}
+
+// Generate generates the configuration image file.
+func (a *ConfigImage) Generate(dependencies asset.Parents) error {
+	ignition := &image.Ignition{}
+
+	dependencies.Get(ignition)
+
+	ca := image.NewCpioArchive()
+
+	// Walk the list of ignition files.
+	// For the ones to be included in the configimage add to cpio archive
+	for _, file := range ignition.Config.Storage.Files {
+		filename := file.Node.Path
+		mode := *file.FileEmbedded1.Mode
+		if includeFileInImage(filename) {
```

Comment:
I have a mixed feeling about keeping the logic of which file should be included in the config ISO here, far away from the Ignition asset: it's in that asset where we effectively decide all the files that will land in the final ignition. Keeping a duplicated list here increases the risk that in future a person will add a new file in the `/data/data/agent/files` folder forgetting to add it here.

Reply:
Yeah its a good point, I originally had this file in `pkg/asset/agent/image/` but all the files there are associated with building an agent image iso so I made a separate dir for this config image. The files that end up in ignition are dispersed though, all of the yamls come from the `manifests` dir.
I don't really like the duplication of files names as it is now but to that point - the files that get included in the config image are there for a reason so I would expect that adding one would be a conscious change and not by pulling in a new file added to ignition.

Let me see if I can find a common place for including this list of files.

Reply:
I've moved the file strings to the ignition asset. Having them there will also allow them to be used to template the load-config script in https://github.com/openshift/installer/pull/7200 so that only these files are copied from the config-image